src/main/java/com/team98/shoppingcart/service/implementation/UserService.java

```diff
@@ -1,15 +1,121 @@
 package com.team98.shoppingcart.service.implementation;
 
+import com.team98.shoppingcart.common.JwtUtil;
+import com.team98.shoppingcart.common.converter.ConvertUtils;
+import com.team98.shoppingcart.config.ApplicationRole;
+import com.team98.shoppingcart.model.entity.Role;
+import com.team98.shoppingcart.model.entity.User;
+import com.team98.shoppingcart.model.request.UserAuthenticationRequest;
+import com.team98.shoppingcart.model.request.UserRegisterRequest;
+import com.team98.shoppingcart.model.response.UserDetailsResponse;
+import com.team98.shoppingcart.model.response.UserRegisterResponse;
+import com.team98.shoppingcart.repository.IUserRepository;
+import com.team98.shoppingcart.service.abstraction.IAuthenticationService;
+import com.team98.shoppingcart.service.abstraction.IUserRegisterService;
+import com.team98.shoppingcart.service.abstraction.IRoleService;
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.EntityNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
-public class UserService implements UserDetailsService {
+public class UserService implements UserDetailsService, IAuthenticationService,
+        IUserRegisterService {
+
+    @Autowired
+    private IUserRepository userRepository;
+
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @Autowired
+    private ConvertUtils convertUtils;
+
+    @Autowired
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Autowired
+    private IRoleService roleService;
 
     @Override
     public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-        return null;
+
+        User user = userRepository.findByEmail(username);
+        if (user == null) {
+            throw new EntityNotFoundException("");
+        }
+        return user;
+    }
+
+    @Override
+    public UserDetailsResponse login(UserAuthenticationRequest authenticationRequest) {
+
+        User user = userRepository.findByEmail(authenticationRequest.getEmail());
+        if (user == null) {
+            throw new EntityNotFoundException("User not found!!!");
```

Comment:
Lo mismo --> throw new UsernameNotFoundException(MessageFormat.format("User {0} not found.", username));

Reply:
ok